File: split/rounddecimals.pas
function rounddecimals(k: smallnumber): scaled;
var
  a: integer;
begin
  a := 0;
  while k > 0 do
  begin
    k := k - 1;
    a := (a + dig[k] * 131072) div 10;
  end;
  rounddecimals := (a + 1) div 2;
end;

File: cpp/rounddecimals.cpp
#include "rounddecimals.h"

scaled rounddecimals(smallnumber k)
{
	int a = 0;
	while (k > 0)
	{
		k--;
		a = (a+dig[k]*131072)/10; // 131'072 = 0x2'00'00
	}
	return (a+1)/2;
}
